src/syscheckd/whodata/audit_healthcheck.c

```diff
@@ -74,25 +77,35 @@ int audit_health_check(int audit_socket) {
     if (audit_delete_rule(AUDIT_HEALTHCHECK_DIR, WHODATA_PERMS, AUDIT_HEALTHCHECK_KEY) <= 0) {
         mdebug1(FIM_HEALTHCHECK_CHECK_RULE); // LCOV_EXCL_LINE
     }
-    hc_thread_active = 0;
+    atomic_int_set(&hc_thread_active, 0);
+
+    // Lock this thread (with 5 seconds timeout) until the healthcheck thread has ended.
+    w_mutex_lock(&audit_hc_mutex);
+    gettime(&wait_time);
+    wait_time.tv_sec += 5;
+    pthread_cond_timedwait(&audit_hc_started, &audit_hc_mutex, &wait_time);
```

Comment:
`audit_hc_started` should be renamed to better reflect what it is used for, it no longer serves as the condition variable to wait for the healthcheck thread to start, it is also used to wait for that same thread to stop. Maybe we could rename it to something like `audit_hc_cond`